### PR TITLE
Undo moving of TurboModule to internal and expose utility function for TurboModule.class.isAssignableFrom

### PR DESCRIPTION
**Pull request by cortinico**
## Summary:

After discussing with [@dvacca](https://www.internalfb.com/intern/profile/?id=1268286274), we prefer to undo the change of TurboModule package to .internal as this is a quite aggressive breaking change for the ecosystem.

Moreover: users should not invoke TurboModule.class.isAssignableFrom because TurboModule is .internal. Therefore I'm exposing another API to check if a class is a TurboModule as a static field of ReactModuleInfo.

## Changelog:

[INTERNAL] - Do not use TurboModule.class.isAssignableFrom

## Test Plan:

Tests are attached

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/module/model/ReactModuleInfo.java [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 package com.facebook.react.module.model;
 
+import com.facebook.react.turbomodule.core.interfaces.TurboModule;
```
Comment (tido64) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/module/model/ReactModuleInfo.java:10]: Shouldn't we be using the `.internal` one? This is what tripped me up as well (and likely others). It'd better if we could make this a build time break instead e.g. if `com.facebook.react.turbomodule.core.interfaces.TurboModule` was removed entirely.
```suggestion
import com.facebook.react.internal.turbomodule.core.interfaces.TurboModule;
```

Comment (tido64) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/module/model/ReactModuleInfo.java:10]: Alternatively, I think you can solve this by using `typealias`? In `com.facebook.react.turbomodule.core.interfaces.TurboModule`:
```kotlin
typealias TurboModule = com.facebook.react.internal.turbomodule.core.interfaces.TurboModule
```

Comment (cortinico) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/module/model/ReactModuleInfo.java:10]: > Shouldn't we be using the .internal one?

Yup we should OR and check both classes actually

**Review by tido64 — APPROVED** [review]
Thanks working on this ❤️

**Pull request closed without merging**